client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx

```diff
@@ -60,10 +60,11 @@ const CommentDialog = ({open, handleDialogClose}: Props) => {
                      step: 'Error occured in adding comment to investigation'
                  });
             })
-            .finally(onDialogClose);
+            .finally(handleDialogClose);
```

Comment:
Don't you need to use onDialogClose?

Reply:
onDialogClose is for when the dialog is closed without saving

Reply:
and so it resets the state

Reply:
when its saved the state doesnt need to reset